### ISSUE TITLE
Bug: [no-deprecated] False positive when destructuring object

### ISSUE THREAD

**Issue by alumni** [typescript-eslint/typescript-eslint#12267]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&fileType=.tsx&code=PQKhCgAIUghBDAzgUwKIDdkDsAukBOyADoSropDgBbKQBGStAxgDZIUBmA9vpPCy0jJM5SlXh46%2BLvAAmTJDmS8ysiji5ja2HAEscAT0iIArnURN8uussgB3Glj5PhOyFyZMT%2BRADooIMDgyAAeRDx4urjKHPBMtAgoGG4A3lCQoBCQ2TAAIhLwAMpc3vGQJiiykFFaQiI4-tnQQdmyBcWlyABcxjhWWADm4OmZ6dCQ%2BThFJfhlFchVNdTa9Y05YzAAArLEhApKVYhRZQCMvgAMF76QAKookCmbLFEA1pBtUx2zyAC%2B1ViIKZYMpRQHIORrZrpJhcLBYZBMPSwnqA-pDH7DFjIPCuXA9RJoeqQAC8D3e7Rm8R6AHJ4NSADSQGFwhFIrA0umQH4AbmGaWyzMBZI%2B006PWZ8MRulhXJJdR0vIFsMQXCxvhYXAGAAoJazpVgAJS8jHgflM5V4XVSmWk3ENEVfeKK80A1XIdWanWwyVso3gDFAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oBNFjpEZAIb5E3dFETRoHaJHBgAviEVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
let event: BaseEvent = { dataSource: 'a', connection: 'a' }; // `connection` is deprecated in `BaseEvent`

{
  const { dataSource: connection } = event; // error
  console.log(connection);
}

{
  const connection = event.dataSource; // no error
  console.log(connection);
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-deprecated": ["error"],
  },
};
```

### tsconfig

```jsonc

```

### Expected Result

I would expect that neither `const { dataSource: connection } = event;` nor `const connection = event.dataSource;` are errors.

### Actual Result

- `const { dataSource: connection } = event;` is an error (false positive);
- `const connection = event.dataSource;` is not an error (correct)

### Additional Info

Noticed while testing `typeorm@1.0`.
